components/focus-trap/focus-trap.js

```diff
@@ -0,0 +1,97 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { isComposedAncestor } from '../../helpers/dom.js';
+
+class FocusTrap extends LitElement {
+
+	static get properties() {
+		return {
+			trap: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
```

Comment:
Nit: include the `:host([hidden]) { display: hidden; }` in case folks want to hide it that way.